code/controllers/subsystem/timer.dm

```diff
@@ -246,7 +246,7 @@ var/datum/subsystem/timer/SStimer
 	if (callBack && callBack.object && callBack.object != GLOBAL_PROC && callBack.object.active_timers)
 		callBack.object.active_timers -= src
 		UNSETEMPTY(callBack.object.active_timers)
-
+		qdel(callBack)
```

Comment:
Also, I should note, that its intended in timer and throwing's case, that the callback can be cached and re-used if needed